File: app/src/main/java/com/example/pizzaHut/viewmodels/HomeFragmentViewModel.java
package com.example.pizzaHut.viewmodels;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.pizzaHut.repositories.HomeFragmentRepository;
import com.example.pizzaHut.responses.CategoryResponse;
import com.example.pizzaHut.responses.IngredientResponse;
import com.example.pizzaHut.responses.MealResponse;

public class HomeFragmentViewModel extends AndroidViewModel {

    private final HomeFragmentRepository repository;

    public HomeFragmentViewModel(Application application) {
        super(application);
        repository = new HomeFragmentRepository();
    }

    public LiveData<MealResponse> getRandomMeal() {
        return repository.getRandomMeal();
    }

    public LiveData<CategoryResponse> getCategories() {
        return repository.getCategories();
    }

    public LiveData<MealResponse> getAreas() {
        return repository.getAreas();
    }

    public LiveData<IngredientResponse> getIngredients() {
        return repository.getIngredients();
    }
}

File: app/src/main/java/com/example/foodRecipes/viewmodels/HomeFragmentViewModel.kt
package com.example.foodRecipes.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.foodRecipes.repositories.HomeRepository
import com.example.foodRecipes.responses.CategoryResponse
import com.example.foodRecipes.responses.MealResponse
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class HomeFragmentViewModel @Inject constructor(private val repository: HomeRepository) : ViewModel() {

    fun getRandomMeal(): LiveData<MealResponse?> = repository.getRandomMeal()

    fun getCategories(): LiveData<CategoryResponse?> = repository.getCategories()

    fun getAreas(): LiveData<MealResponse?> = repository.getAreas()
}